File: Codeforces/Div4/784/Fall-Down/main.py
# https://codeforces.com/contest/1669/problem/G
# Time limit exceeded

import io, os, sys

input = io.BytesIO(os.read(0, os.fstat(0).st_size)).readline

STONE = '*'
EMPTY = '.'

def readline():
    return input().decode()

def readint():
    return int(readline())

def readcharlist():
    return list(readline())
 
def can_move (x, y, n, grid):
    return x+1 < n and grid[x][y] == STONE and grid[x+1][y] == EMPTY
 
def simulate(n, m, grid):
    last_line_moved = n-2
    did_move = 1
 
    while did_move:
        did_move = 0

        for i in range(last_line_moved, -1, -1):
            for j in range(m):
                if can_move(i, j, n, grid):
                    did_move = 1
                    tup = grid[i][j], grid[i+1][j]
                    grid[i+1][j], grid[i][j] = tup
            last_line_moved += int(did_move)
    return grid
 
def print_grid(grid):
    for line in grid:
        for c in line:
            sys.stdout.write(c)
    sys.stdout.write('\n')
    return 
 
def main ():
    for i in range(readint()):
        grid = []
        n, m = [int(x) for x in readline().split()]
 
        for j in range(n):
            grid.append(readcharlist())
        
        simulate(n, m, grid)
        print_grid(grid)
 
main()

File: Codeforces/Div4/784/Fall-Down/main.cpp
// https://codeforces.com/contest/1669/problem/G

#include<bits/stdc++.h>

#define EMPTY '.'
#define STONE '*'
#define OBSTACLE 'o'

using namespace std;

int n, m, t;
char grid[50][50];

bool canMoveStone (int x, int y) {
    return grid[x+1][y] == EMPTY && grid[x][y] == STONE;  
}

void simulate () {
    int i, j;
    for(i=0; i<n; i++)
        for(j=0; j<m; j++)
            cin >> grid[i][j];

    bool didMove = true;

    while (didMove) {
        int lastFileMoved = n-2;
        didMove = false;
        for (i=lastFileMoved; i>=0; i--) { 
            for (j=0; j<m; j++) {
                if (canMoveStone(i, j)) {
                    didMove = true;
                    swap(grid[i][j], grid[i+1][j]);
                }
            }

            if (!didMove)
               lastFileMoved--; 
        }
    }
}

void printResult() {
    for(int i=0; i<n; i++) {
        for(int j=0; j<m; j++)
            cout << grid[i][j];
        cout << endl;
    }
    cout << endl;
}

int main () {
    cin.tie(nullptr);
    ios_base::sync_with_stdio(false);

    cin >> t;

    while(t--) {
        cin >> n >> m;
        simulate();
        printResult();
    }   

    return 0;
}